src/Models/Context/SceneThemeContext/SceneThemeContext.tsx

```diff
@@ -0,0 +1,231 @@
+        logDebugConsole(
+            'info',
+            `Updating Color context ${action.type} with payload: `,
+            action.payload
+        );
+        switch (action.type) {
+            case SceneThemeContextActionType.SET_OBJECT_COLOR: {
+                draft.objectColor =
+                    action.payload.color ||
+                    draft.objectColorOptions?.[0]?.color ||
+                    '';
+                setPersistedTheme(buildPersistedTheme(draft));
+                break;
+            }
+            case SceneThemeContextActionType.SET_OBJECT_COLOR_OPTIONS: {
+                draft.objectColorOptions = action.payload.options || [];
+                break;
+            }
+            case SceneThemeContextActionType.SET_OBJECT_STYLE: {
+                draft.objectStyle =
+                    action.payload.style || ViewerObjectStyle.Default;
+                if (draft.objectStyle === ViewerObjectStyle.Default) {
+                    draft.objectColor =
+                        draft.objectColorOptions[
+                            DEFAULT_OBJECT_COLOR_INDEX
+                        ].color; // white
+                } else if (
+                    draft.objectStyle === ViewerObjectStyle.Transparent &&
+                    draft.objectColor ===
+                        draft.objectColorOptions[DEFAULT_OBJECT_COLOR_INDEX]
+                            .color // if it's white, change it cause white doesn't work so we want the default to be nice
+                ) {
+                    draft.objectColor = draft.objectColorOptions[0].color; // blue
+                }
+                setPersistedTheme(buildPersistedTheme(draft));
+                break;
+            }
+            case SceneThemeContextActionType.SET_OBJECT_STYLE_OPTIONS: {
+                draft.objectStyleOptions = action.payload.options || [];
+                break;
+            }
+            case SceneThemeContextActionType.SET_SCENE_BACKGROUND: {
+                draft.sceneBackground =
+                    action.payload.color ||
+                    draft.sceneBackgroundOptions?.[0]?.color ||
+                    '';
+                setPersistedTheme(buildPersistedTheme(draft));
+                break;
+            }
+            case SceneThemeContextActionType.SET_SCENE_BACKGROUND_OPTIONS: {
+                draft.sceneBackgroundOptions = action.payload.options || [];
+                break;
+            }
+        }
+    }
+);
+
+export const SceneThemeContextProvider: React.FC<ISceneThemeContextProviderProps> = (
+    props
+) => {
+    const { children } = props;
+
+    // skip wrapping if the context already exists
+    const existingContext = useSceneThemeContext();
+    if (existingContext) {
+        return <>{children}</>;
+    }
+
+    const { initialState, usePersistedTheme = true } = props;
+    const { t } = useTranslation();
+    shouldPersistTheme = usePersistedTheme; // store the selection globally
```

Comment:
nit: starting a variable name with `use` is unofficially reserved for react hooks.  Maybe rename to `isThemePersisted`?

Reply:
Fair. Updated.